Gems/AtomLyIntegration/CommonFeatures/Assets/Scripts/Profiling/OutputProfileData.lua

```diff
@@ -0,0 +1,134 @@
+local SourceProjectUserPathRegistryKey <const> = "/O3DE/Runtime/FilePaths/SourceProjectUserPath"
+local ConsoleCommandQuitRegistryKey <const> = "/Amazon/AzCore/Runtime/ConsoleCommands/quit"
+
+function OutputProfileData:TryDisconnect()
+    if (self.active) then
+        self.active = false
+        self.profileCaptureNotificationHandler:Disconnect()
+        self.tickHandler:Disconnect()
+    end
+end
+
+function OutputProfileData:TryQuitOnComplete()
+    if (self.quitOnComplete) then
+        g_SettingsRegistry:SetString(ConsoleCommandQuitRegistryKey, "")
+    end
+end
+
+function OutputProfileData:TryCapture()
+    self.captureInProgress = true
+    self.cpuTimingsOutputPath = tostring(self.outputFolder) .. '/cpu_frame' .. tostring(self.captureCount) .. '_time.json'
+    ProfilingCaptureRequestBus.Broadcast.CaptureCpuFrameTime(self.cpuTimingsOutputPath)
+end
+
+function OutputProfileData:CaptureFinished()
+    self.captureInProgress = false
+end
+
+
+function OutputProfileData:OnActivate()
+    if (g_SettingsRegistry:IsValid()) then
+        local quitOnCompleteValue = g_SettingsRegistry:GetBool(QuitOnCompleteRegistryKey)
+        self.quitOnComplete = quitOnCompleteValue:value_or(self.Properties.QuitOnComplete)
+
+        local frameTimeRecordingActivateValue = g_SettingsRegistry:GetBool(FrameTimeRecordingActiveRegistryKey)
+        if (not frameTimeRecordingActivateValue:has_value() or not frameTimeRecordingActivateValue:value()) then
+            Debug:Log("OutputProfileData:OnActivate - Missing registry setting to activate frame time recording, aborting data collection")
+            self:TryQuitOnComplete()
+            return
+        end
+
+        -- get path to user folder
+        local pathToUserFolder = "InvalidPath/"
+        local settingsRegistryResult =  g_SettingsRegistry:GetString(SourceProjectUserPathRegistryKey)
+        if (settingsRegistryResult:has_value()) then
+            pathToUserFolder = settingsRegistryResult:value()
+        else
+            Debug:Log("OutputProfileData:OnActivate - Unable to resolve the SourceProjectUserPath, aborting data collection")
+            self:TryQuitOnComplete()
+            return
+        end
+
+        -- get any registry property overrides
+        local frameDelayCountValue = g_SettingsRegistry:GetUInt(FrameDelayCountRegistryKey)
+        self.frameDelayCount = frameDelayCountValue:value_or(self.Properties.FrameDelayCount)
+        local frameCaptureCountValue = g_SettingsRegistry:GetUInt(FrameCaptureCountRegistryKey)
+        self.frameCaptureCount = frameCaptureCountValue:value_or(self.Properties.FrameCaptureCount)
+        local profileNameValue = g_SettingsRegistry:GetString(ProfileNameRegistryKey)
+        self.profileName = profileNameValue:value_or(self.Properties.ProfileName)
+
+        -- generate final output path string
+        self.outputFolder = tostring(pathToUserFolder) .. "/Scripts/PerformanceBenchmarks/" .. tostring(self.profileName)
```

Comment:
"Scripts" seems like a strange name to be part of the output folder. The result of running a script is a frame capture file
It seems like a name of "log/PerformanceBenchmarks or "log/FrameCapture" would be more appropriate.

We also have a Settings Registry key for the log folder path.
[/O3DE/Runtime/FilePaths/SourceProjectLogPath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.h#L53)

Reply:
Log seems reasonable to me, I used scripts because that's the existing behavior for the AtomSampleViewer profiling data.

Reply:
The user/scripts/performanceBenchmarks path comes from the data aggregator that is used for the performance benchmarks.